From b7ca9dd97bbd9657c541f749ea6baf1f45b7c98a Mon Sep 17 00:00:00 2001
From: Thomas Petazzoni <thomas.petazzoni@bootlin.com>
Date: Wed, 9 Sep 2020 22:22:18 +0200
Subject: [PATCH] libcap/Makefile: disable building/installing shared
 library when DYNAMIC is empty

Signed-off-by: Thomas Petazzoni <thomas.petazzoni@bootlin.com>
[slightly manual modifications that patch still applies for libcap 2.44]
Signed-off-by: Heiko Thiery <heiko.thiery@gmail.com>
---
 libcap/Makefile | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/libcap/Makefile b/libcap/Makefile
index 81b089e..dfd4dea 100644
--- a/libcap/Makefile
+++ b/libcap/Makefile
@@ -22,7 +22,7 @@ MAJLIBNAME=$(LIBNAME).$(VERSION)
 MINLIBNAME=$(MAJLIBNAME).$(MINOR)
 GPERF_OUTPUT = _caps_output.gperf
 
-all: $(MINLIBNAME) $(STACAPLIBNAME) pcs $(STAPSXLIBNAME)
+all: $(if $(DYNAMIC),$(MINLIBNAME)) $(STACAPLIBNAME) pcs $(STAPSXLIBNAME)
 
 pcs: libcap.pc libpsx.pc
 
@@ -93,7 +93,7 @@ cap_test: cap_test.c libcap.h
 test: cap_test
 	./cap_test
 
-install: install-static install-shared
+install: $(if $(DYNAMIC),install-shared) install-static
 
 install-common: pcs
 	mkdir -p -m 0755 $(FAKEROOT)$(INCDIR)/sys
-- 
2.26.2

